libraries/GCS_MAVLink/GCS_Common.cpp

```diff
@@ -1650,7 +1662,17 @@ MAV_RESULT GCS_MAVLINK::handle_preflight_reboot(const mavlink_command_long_t &pa
             }
 #endif
         }
-
+        if (should_zero_rc_outputs_on_reboot()) {
+            reboot_zero_rc_outputs();
+        }
+        // send ack before we reboot
+        mavlink_msg_command_ack_send(chan, packet.command, MAV_RESULT_ACCEPTED);
```

Comment:
we should make sure to flush the tx buffers. Relying on a delay below to do that is not good IMO.

Reply:
Is there a good way to do that from the main thread, 'though?  This is a no-worse-off-and-possibly-better patch as-is.